src/csvcubed/inspect/lazyfuncdescriptor.py

```diff
@@ -0,0 +1,52 @@
+from dataclasses import field
+from typing import Any, Callable, Generic, Mapping, Optional, Type, TypeVar
+
+TClass = TypeVar("TClass")
+TRet = TypeVar("TRet")
+
+
+class LazyFuncFieldDescriptor(Generic[TClass, TRet]):
+    """
+    https://docs.python.org/3/library/dataclasses.html#descriptor-typed-fields
```

Comment:
This could use a better explanation of why it's here.

Also, could you check each of the modules you've created and ensure they all have a docstring at the top of the module according to our [dev standards](https://github.com/GSS-Cogs/csvcubed/blob/main/docs/developer.md#api-documentation)?

Reply:
The docstrings have been added to the top of all the new files we made.